src/graph.rs

```diff
@@ -532,12 +532,30 @@ impl Serialize for Value {
             Value::Boolean(value) => serializer.serialize_bool(*value),
             Value::Integer(value) => serializer.serialize_u32(*value),
             Value::String(value) => serializer.serialize_str(value),
-            // FIXME: there's no way to distinguish sets and lists, so we can't roundtrip accurately
-            Value::List(list) => serializer.collect_seq(list),
-            Value::Set(set) => serializer.collect_seq(set),
-            // FIXME: we don't distinguish between syntax tree node IDs, graph node IDs, and integers
-            Value::SyntaxNode(node) => serializer.serialize_u32(node.index),
-            Value::GraphNode(node) => serializer.serialize_u32(node.0),
+            Value::List(list) => {
+                let mut map = serializer.serialize_map(None)?;
+                map.serialize_entry("type", "list")?;
+                map.serialize_entry("value", list)?;
+                map.end()
+            }
```

Comment:
The format seems unnecessarily sparse, but it matches what we've done elsewhere. If we were to tighten it up, I'd suggest `{ "list": […] }`.

Reply:
It also seems like we could leave this one using arrays and only give sets special formatting.

Reply:
I don't _think_ further fields are likely, as those would likely be annotated via attrs on nodes/edges rather than on the values making them up. (I think?) But it's a point to consider.